editor/src/core/model/element-metadata-utils.ts

```diff
@@ -1695,24 +1718,50 @@ function fillSpyOnlyMetadataWithFramesFromChildren(
     }
 
     const spyElem = fromSpy[pathStr]
-    const childrenFromSpy = MetadataUtils.getChildren(fromSpy, spyElem.elementPath)
-    const childrenFromDom = MetadataUtils.getChildren(fromDOM, spyElem.elementPath)
-    const childrenNotInDom = childrenFromSpy.filter((childNotInDom) =>
-      childrenFromDom.every(
-        (childInDom) => !EP.pathsEqual(childNotInDom.elementPath, childInDom.elementPath),
-      ),
+
+    const { children: childrenFromSpy, unfurledComponents: unfurledComponentsFromSpy } =
+      MetadataUtils.getAllChildrenElementsIncludingUnfurledFocusedComponents(
+        spyElem.elementPath,
+        fromSpy,
+      )
+    const childrenAndUnfurledComponentsFromSpy = [...childrenFromSpy, ...unfurledComponentsFromSpy]
+
+    const { children: childrenFromDom, unfurledComponents: unfurledComponentsFromDom } =
+      MetadataUtils.getAllChildrenElementsIncludingUnfurledFocusedComponents(
+        spyElem.elementPath,
+        fromDOM,
+      )
+    const childrenAndUnfurledComponentsFromDom = [...childrenFromDom, ...unfurledComponentsFromDom]
+
+    const childrenAndUnfurledComponentsNotInDom = childrenAndUnfurledComponentsFromSpy.filter(
+      (childNotInDom) =>
+        childrenAndUnfurledComponentsFromDom.every(
+          (childInDom) => !EP.pathsEqual(childNotInDom.elementPath, childInDom.elementPath),
+        ),
     )
-    const recursiveChildren = childrenNotInDom.flatMap((c) => {
-      return findChildrenInDomRecursively(EP.toString(c.elementPath))
-    })
-    const children = [...childrenFromDom, ...recursiveChildren]
 
-    childrenInDomCache[pathStr] = children
+    const recursiveChildrenAndUnfurledComponents = childrenAndUnfurledComponentsNotInDom.flatMap(
+      (c) => {
+        return findChildrenInDomRecursively(EP.toString(c.elementPath))
+      },
+    )
+
+    const childrenAndUnfurledComponents = [
+      ...childrenAndUnfurledComponentsFromDom,
+      ...recursiveChildrenAndUnfurledComponents,
+    ]
+
+    childrenInDomCache[pathStr] = childrenAndUnfurledComponents
 
-    return children
+    return childrenAndUnfurledComponents
   }
 
   const elementsWithoutDomMetadata = Object.keys(fromSpy).filter((p) => fromDOM[p] == null)
+  // Sort and then reverse these, so that lower level elements are handled ahead of their parents
+  // and ancestors. This means that if there are a grandparent and parent which both lack global frames
+  // then the parent is fixed ahead of the grandparent, which will be based on the parent.
+  elementsWithoutDomMetadata.sort()
```

Comment:
I _think_ if there is a UID in the dataset that is more than 3 letters long, the sort will not be correct anymore! (as far as I see this is sorting an array of stringified elementPaths, implicitly assuming that shorter string = higher level element)